util/src/test/java/io/camunda/zeebe/util/FileUtilTest.java

```diff
@@ -90,4 +97,60 @@ void shouldCopySnapshot() throws Exception {
     // then
     assertThat(Files.list(target)).contains(target.resolve(snapshotFile));
   }
+
+  @Test
+  void shouldPreallocateFile() throws IOException {
+    // given
+    final var path = tmpDir.resolve("file");
+    final var length = 1024 * 1024L;
+
+    // when
+    try (final FileChannel channel =
+        FileChannel.open(path, StandardOpenOption.CREATE, StandardOpenOption.WRITE)) {
+      FileUtil.preallocate(channel, length);
+    }
+
+    // then
+    final var realSize = getRealSize(path);
+    assertThat(realSize)
+        .as(
+            "Expected <%s> to have a real size of <%d> bytes, but it had <%d>",
+            path, length, realSize)
+        .isEqualTo(length);
```

Comment:
Yes, you're 100% right, I thought about it, but was just very lazy :see_no_evil: It works as is as long as the size is a multiple of your I/O block size :sweat_smile:

Reply:
Ah, I remember, the main issue is that you'll be between the real size and at most one block most. If we assume block size is 4096, all good - if we don't though, then this will fail. So it's a little better, but I guess still not really accurate. 

I could push it down and instead of `getRealSize` (which is more flexible), we just have a `assertThatFileHasRealSize()` and `assertThatFileHasRealSizeLessThan()`. When checking for equality, we can reuse the `FileStat#blockSize` to get an accurate upper bound.

Reply:
Actually, I just added a `getBlockSize`, it was less code in the end :shrug:

Reply:
Nice! Sorry, I didn't realize that this would be that much work, I just thought a `.isGreaterThan(length)` would have been enough but your version is obviously much stricter and better :+1: